### PR TITLE
Cleanup Kconfig from board DTS

### PR DESCRIPTION
**Pull request by nashif**
Move to overlays and remove configurable features from boards.

### CODE REVIEW

File: boards/x86/tinytile/tinytile.dts [hunk 1]
```diff
@@ -20,11 +20,7 @@
 	chosen {
 		zephyr,sram = &sram0;
 		zephyr,flash = &flash0;
-#ifdef CONFIG_USB_UART_CONSOLE
-		zephyr,console = &usb_cdc;
-#else
 		zephyr,console = &uart1;
-#endif
 		zephyr,bt-uart = &uart0;
```
Comment (finikorg) [inline boards/x86/tinytile/tinytile.dts]: Does the board has only USB Serial console only? How the board works after that?

File: samples/subsys/usb/console/tinytile.overlay [hunk 2]
```diff
@@ -0,0 +1,5 @@
+/ {
+        chosen {
+                zephyr,console = &usb_cdc;
+        };
+};
```
Comment (finikorg) [inline samples/subsys/usb/console/tinytile.overlay]: How to include these overlays to build?

**Review by galak — DISMISSED** [review]
If you are removing CONFIG_BT, we should remove CONFIG_MODEM_WNCM14A2A

**Comment by nashif** [comment]
> If you are removing CONFIG_BT, we should remove CONFIG_MODEM_WNCM14A2A

will come in another PR, there is no dependency between the two.

File: samples/subsys/usb/console/tinytile.overlay (hunk 2 above)
Comment (nashif) [reply samples/subsys/usb/console/tinytile.overlay]: this is included by the build system directly.

File: boards/x86/tinytile/tinytile.dts (hunk 1 above)
Comment (nashif) [reply boards/x86/tinytile/tinytile.dts]: > Does the board has only USB Serial console only?

no, it does have UART.

Board will work exactly the same way, if you want to use CDC, then you will need an overlay.

**Comment by nashif** [comment]
> If you are removing CONFIG_BT, we should remove CONFIG_MODEM_WNCM14A2A

ok, added a commit now, but this should end up in a shield IMO. Filed issue #10965.

**Comment by galak** [comment]
Looks good to me, want to let it sit around for a bit for @mike-scott and @MaureenHelm to have a chance to comment.

File: boards/arm/frdm_k64f/dts_fixup.h [hunk 3]
```diff
@@ -13,19 +13,3 @@
 #define CONFIG_MCR20A_GPIO_IRQ_B_PIN			NXP_KINETIS_DSPI_4002C000_NXP_MCR20A_0_IRQB_GPIOS_PIN
 #define CONFIG_MCR20A_GPIO_RESET_NAME			NXP_KINETIS_DSPI_4002C000_NXP_MCR20A_0_RESET_GPIOS_CONTROLLER
 #define CONFIG_MCR20A_GPIO_RESET_PIN			NXP_KINETIS_DSPI_4002C000_NXP_MCR20A_0_RESET_GPIOS_PIN
```
Comment (finikorg) [inline boards/arm/frdm_k64f/dts_fixup.h]: Should we do the same for MCR20A?

**Review by MaureenHelm — CHANGES_REQUESTED** [review]
@nashif I guess you're all in on dts overlays now? 😄 I'll look at adding a bt shield overlay for kw41

File: boards/arm/hexiwear_k64/hexiwear_k64.dts [hunk 4]
```diff
@@ -37,9 +37,6 @@
 		zephyr,sram = &sram0;
 		zephyr,flash = &flash0;
 		zephyr,console = &uart0;
-#ifdef CONFIG_BT
-		zephyr,bt-uart = &uart4;
```
[the remark this answers is not in the document]
Comment (nashif) [reply boards/arm/hexiwear_k64/hexiwear_k64.dts]: ok, we can just remove the ifdef then.

**Comment by nashif** [comment]
> @nashif I guess you're all in on dts overlays now?

not really, but we have to move forward, we will find a way to make them user-friendly soon enough :)

**Comment by mike-scott** [comment]
@nashif I'm ok with the modem removal here.  I'll set it up as a shield in a new PR shortly.

File: boards/arm/frdm_k64f/frdm_k64f.dts [hunk 5]
```diff
@@ -119,33 +116,10 @@
 	current-speed = <115200>;
 };
 
-#ifdef CONFIG_UART_MCUX_2
 &uart2 {
```
Comment (galak) [inline boards/arm/frdm_k64f/frdm_k64f.dts]: uart2 should be removed here, this was done just for the Modem

File: boards/arm/hexiwear_k64/hexiwear_k64.dts (hunk 4 above)
[the remark this answers is not in the document]
Comment (nashif) [reply boards/arm/hexiwear_k64/hexiwear_k64.dts]: leaving that was causing build issues, because uart4 is not defined?

**Review by MaureenHelm — CHANGES_REQUESTED** [review]

File: tests/drivers/build_all/testcase.yaml [hunk 6]
```diff
@@ -13,7 +13,7 @@ tests:
     min_ram: 32
     platform_exclude: zedboard_pulpino frdm_kw41z
     tags: drivers footprint
-    depends_on: adc spi
+    depends_on: adc spi netif
```
[the remark this answers is not in the document]
Comment (nashif) [reply tests/drivers/build_all/testcase.yaml]: fixed

File: boards/arm/hexiwear_k64/hexiwear_k64.dts [hunk 7]
```diff
@@ -37,9 +37,7 @@
 		zephyr,sram = &sram0;
 		zephyr,flash = &flash0;
 		zephyr,console = &uart0;
-#ifdef CONFIG_BT
```
Comment (MaureenHelm) [inline boards/arm/hexiwear_k64/hexiwear_k64.dts]: The changes here are fine, but the commit message doesn't make sense anymore. Will you please update the commit message?

**Review by MaureenHelm — CHANGES_REQUESTED** [review]

**Review by MaureenHelm — APPROVED** [review]

**Pull request merged**